run/run.py

```diff
@@ -254,6 +244,29 @@ def main(platform_id, platform, args, config):
     print('Response text:', response.text)
 
 
+def setup_wpt(mainargs, platform, config, logger):
+    wpt_setup_commands = [
+        ['git', 'reset', '--hard', 'HEAD'],  # For wpt.patch
+        ['git', 'checkout', 'master'],
+        ['git', 'pull'],
+        ['./wpt', 'manifest', '--work'],
+    ]
+    for command in wpt_setup_commands:
+        return_code = subprocess.check_call(command, cwd=config['wpt_path'])
+        assert return_code == 0, (
+            'Got non-0 return code: '
+            '%d from command %s' % (return_code, command))
+
+    patch_wpt(config, platform)
+
+    if mainargs.wpt_sha:
+        return mainargs.wpt_sha
+    else:
+        sha_finder = shas.SHAFinder(logger)
+        return (sha_finder.get_todays_sha(config['wpt_path'])
+                or sha_finder.get_head_sha(config['wpt_path']))
```

Comment:
FWIW, this bit won't make sense if the `git checkout master; git pull` is changed. It'd work if `get_head_sha` did `git rev-parse origin/HEAD` instead of `git rev-parse HEAD`.